trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/file/FileBasedCredentialsProvider.java

```diff
@@ -38,7 +39,7 @@ public class FileBasedCredentialsProvider
     private final Map<String, Credentials> credentialsStore;
 
     @Inject
-    public FileBasedCredentialsProvider(FileBasedCredentialsProviderConfig config, ObjectMapper objectMapper)
+    public FileBasedCredentialsProvider(FileBasedCredentialsProviderConfig config, @Named("FileBasedCredentials") ObjectMapper objectMapper)
```

Comment:
We use custom binding annotations instead of `@Named`. See `@ForProxyClient` as an example